eventlet/patcher.py

```diff
@@ -90,6 +90,10 @@ def inject(module_name, new_globals, *additional_modules):
     # Remove the old module from sys.modules and reimport it while
     # the specified modules are in place
     sys.modules.pop(module_name, None)
+    # Also remove sub modules and reimport
+    for imported_module_name in list(sys.modules.keys()):
```

Comment:
Yeah, I believe a test case is necessary here, the monkey patching mechanism is tricky so it'd help keeping it working correctly in the future.

Reply:
Added unit test and comments

Reply:
Since there's some global state involved I think making this an isolated test would be better. See `tests/patcher_test.py` for examples – in general you just add a simple test function calling `run_isolated('xxx.py')` and move your actual test into `tests/isolated/xxx.py`

Reply:
Done